sse2neon.h

```diff
@@ -6368,6 +6368,11 @@ FORCE_INLINE int _mm_testz_si128(__m128i a, __m128i b)
 #define _mm_extract_epi16(a, imm) \
     vgetq_lane_u16(vreinterpretq_u16_m128i(a), (imm))
 
+// Extract a 16-bit integer from a, selected with imm8, and store the result in
+// the lower element of dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_extract_pi16
+#define _mm_extract_pi16(a, imm) vget_lane_u16(vreinterpret_u16_m64(a), (imm))
```

Comment:
Use inline function instead. Thus, checks on parameters can be applied.

Reply:
I met an error 
```
./sse2neon.h:6376:12: error: argument to '__builtin_neon_vget_lane_i16' must be a constant integer
    return vget_lane_u16(vreinterpret_u16_m64(a), imm);
```
So I change it back to macro

Reply:
Use multi-line form of macro. See `_mm_srai_epi32` as an example.
The single-line form of macro is used for equivalent expression.